### PR TITLE
 Move Media into the ScoreBoard and use inotify.

### PR DESCRIPTION
**Pull request by brian-brazil**
This gets us a bit closer to #75

### CODE REVIEW

File: src/com/carolinarollergirls/scoreboard/defaults/DefaultMediaModel.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/defaults/DefaultMediaModel.java]: `Locale` is unused

File: src/com/carolinarollergirls/scoreboard/jetty/MediaServlet.java [hunk 2]
```diff
@@ -38,17 +38,10 @@
 import org.apache.commons.io.filefilter.IOFileFilter;
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/jetty/MediaServlet.java]: `MalformedURLException`, `URL`, `Map`, `ConcurrentHashMap`, `IOFileFilter`, and `XmlDocumentEditor` are unused.

File: src/com/carolinarollergirls/scoreboard/xml/ScoreBoardXmlListener.java [hunk 3]
```diff
@@ -20,6 +20,7 @@
 import com.carolinarollergirls.scoreboard.event.ScoreBoardListener;
 import com.carolinarollergirls.scoreboard.model.SettingsModel;
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/xml/ScoreBoardXmlListener.java]: `SettingsModel` is unused. (Probably left over from an older PR.)

File: tests/com/carolinarollergirls/scoreboard/defaults/DefaultRulesetsModelTests.java [hunk 4]
```diff
@@ -25,7 +25,6 @@
 
     @Before
     public void setUp() throws Exception {
-        ScoreBoardManager.setPropertyOverride(JettyServletScoreBoardController.class.getName() + ".html.dir", "html");
```
Comment (frank-weinberg) [inline tests/com/carolinarollergirls/scoreboard/defaults/DefaultRulesetsModelTests.java:28]: This makes `ScoreBoardManager` and `JettyServletScoreBoardController` unused.

File: tests/com/carolinarollergirls/scoreboard/defaults/DefaultScoreboardModelTests.java [hunk 5]
```diff
@@ -59,7 +59,6 @@ public void scoreBoardChange(ScoreBoardEvent event) {
     @Before
     public void setUp() throws Exception {
         ScoreBoardClock.getInstance().stop();
-        ScoreBoardManager.setPropertyOverride(JettyServletScoreBoardController.class.getName() + ".html.dir", "html");
```
Comment (frank-weinberg) [inline tests/com/carolinarollergirls/scoreboard/defaults/DefaultScoreboardModelTests.java:62]: This makes `ScoreBoardManager` and `JettyServletScoreBoardController` unused.

File: tests/com/carolinarollergirls/scoreboard/defaults/DefaultStatsModelTests.java [hunk 6]
```diff
@@ -31,7 +31,6 @@
     public void setUp() throws Exception {
         ScoreBoardClock.getInstance().stop();
 
-        ScoreBoardManager.setPropertyOverride(JettyServletScoreBoardController.class.getName() + ".html.dir", "html");
```
Comment (frank-weinberg) [inline tests/com/carolinarollergirls/scoreboard/defaults/DefaultStatsModelTests.java:34]: This makes `ScoreBoardManager` and `JettyServletScoreBoardController` unused.

File: tests/com/carolinarollergirls/scoreboard/json/ScoreBoardJSONListenerTests.java [hunk 7]
```diff
@@ -35,8 +44,15 @@ public void sendUpdates(Map<String, Object> s, Set<String> changed) {
...
+        oldDir = ScoreBoardManager.getDefaultPath();
+        ScoreBoardManager.setDefaultPath(dir.getRoot());
+        dir.newFolder("config", "penalties");
+        Files.copy(oldDir.toPath().resolve("config/penalties/wftda2018.json"),
+                   dir.getRoot().toPath().resolve("config/penalties/wftda2018.json"));
+        dir.newFolder("html", "images", "teamlogo");
+        dir.newFile("html/images/teamlogo/init.png");
+
         ScoreBoardClock.getInstance().stop();
-        ScoreBoardManager.setPropertyOverride(JettyServletScoreBoardController.class.getName() + ".html.dir", "html");
```
Comment (frank-weinberg) [inline tests/com/carolinarollergirls/scoreboard/json/ScoreBoardJSONListenerTests.java:39]: This makes `JettyServletScoreBoardController` unused (but this time not ScoreBoardManager).

File: src/com/carolinarollergirls/scoreboard/defaults/DefaultMediaModel.java [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+        public String getProviderId() { return getId(); }
+
+        public String getFormat() { synchronized (coreLock) { return format ;} }
+        public String getType() { synchronized (coreLock) { return type ;} }
+        public String getId() { synchronized (coreLock) { return id ;} }
+        public String getName() { synchronized (coreLock) { return name ;} }
+        public void setName(String n) {
+            synchronized (coreLock) {
+                name = n;
+                scoreBoardChange(new ScoreBoardEvent(this, EVENT_FILE, this, null));
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/defaults/DefaultMediaModel.java:223]: This should probably have it's own event, so we aren't forced to replace the whole element in order to change one attribute.

Comment (brian-brazil) [reply src/com/carolinarollergirls/scoreboard/defaults/DefaultMediaModel.java:223]: It's one tr, and not on a critical path. I don't think it's worth it.

**Comment by brian-brazil** [comment]
I've removed the unused imports, looks like we could use checkstyle to catch that automatically.

**Comment by frank-weinberg** [comment]
I get them marked automatically by Eclipse.

**Comment by brian-brazil** [comment]
I use vim, so a bit tricker

**Pull request merged**